### DISCUSSION TITLE
Are these features are available? Unique rows, Disable some CRUD operations

### DISCUSSION THREAD

**Discussion by vacom** [supabase/supabase#713] [Questions] [answered]
I'm still new to the platform and I'm exploring if the following points are possible: 

- Set unique rows based on a column;
- Disable CRUD operations based on table (example: Allow Select, Insert, Update but disable delete)
- Create a new row on Table 2 when inserting on Table 1;
- Prevent some columns from being inserted or modified (example: allow admin role but don't allow user)

**Comment by steve-chavez** [comment, marked as answer]
Yes, all of those are standard capabilities of PostgreSQL.

> Set unique rows based on a column;

UNIQUE constraint: https://www.postgresqltutorial.com/postgresql-unique-constraint/

> Disable CRUD operations based on table (example: Allow Select, Insert, Update but disable delete)

GRANT: https://www.postgresqltutorial.com/postgresql-administration/postgresql-grant/

> Create a new row on Table 2 when inserting on Table 1;

TRIGGER: https://www.postgresqltutorial.com/creating-first-trigger-postgresql/

> Prevent some columns from being inserted or modified (example: allow admin role but don't allow user)

Also GRANT. You can do GRANTs on certain columns or on certain roles(like admin or webuser).

**Reply by vacom** [reply]
Hey @steve-chavez , thanks for the answer, I should have mentioned if this could be possible to do visually with the dashboard but for me with PostgreSQL works as well.  I will mark this answer as the correct one as it is a way to get what I asked.
